tensorflow_lite_support/c/task/common_error_utils.cc

```diff
@@ -0,0 +1,86 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_lite_support/c/task/common_error_utils.h"
+
+#include <string>
+
+#include "external/com_google_absl/absl/strings/cord.h"
+#include "tensorflow_lite_support/cc/common.h"
+
+namespace tflite {
+namespace task {
+
+namespace {
+using ::tflite::support::kTfLiteSupportPayload;
+}  // namespace
+
+void CreateTfLiteErrorWithStatus(const absl::Status& status,
+                                 TfLiteError** error) {
+  if (status.ok() or error == nullptr) return;
+
+  // Payload of absl::Status created by the tflite task library stores an
+  // appropriate value of the enum TfLiteSupportStatus. The integer value
+  // corresponding to the TfLiteSupportStatus enum stored in the payload is
+  // extracted here to later map to the appropriate error code to be returned.
+  // In cases where the enum is not stored in (payload is NULL or the payload
+  // string cannot be converted to an integer), we set the error code value to
+  // be 1 (kError of TfLiteErrorCode used in the C library to signify any errors
+  // not falling into other categories.) Since payload is of type absl::Cord
+  // that can be type cast into an absl::optional<std::string>, we use the
+  // std::stoi function to convert it into an integer code if possible.
+  int generic_error_code = static_cast<int>(kError);
+  int error_code;
+  try {
```

Comment:
@lu-wang-g The C++ layer stores `TfLiteSupportStatus` codes as the payload of `absl::Status` and hence have to be converted back to the specific integers which are later converted to `TfLiteErrorCode` based on whether the integer falls into the range of the `TfLiteErrorCode` enum. status code of `absl::Status` cannot be used to define TfLiteErrorCode since it is not `TfLite Support `specific.
In case of some errors (eg:, When roi is outside image bounds,) the C++ layer does not create the status I guess, it directly passes whatever is thrown by the Tflite Interpreter ? (That's what I gather). In such cases payload does not contain the appropriate `TfLiteSupportStatus code` . So I am assigning a generic error code (`kError`). But the `absl::Status::code()` in such cases has a more meaningful code like 'Invalid Argument'. Should the absl status code be taken into consideration in cases where payload is missing to assign a `TfLiteErrorCode`?. But this will be more complicated, since we will have to monitor the absl status codes thrown by TfLite in different scenarios and define corresponding error codes.

Reply:
That's true that quite a few Status objects are not created with TfLiteSupportStatus. You can match the following common absl:StatusCode: `kInvalidArgument`, `kInternal`, `kNotFound`, and map the rest to kError.

Reply:
I have incorporated this. Added 2 new codes for kInternal and kNotFound in TfLiteSupportStatus since, there are multiple not found codes in TfLiteSupportStatus and nothing that meaningfully signifies kInternal. I have started these from 900 leaving space space if more codes get added to TfLiteSupportStatus and thereby in TfLiteSupportErrorCode. Also all range checks for mapping to TfLiteSupportStatus will still be performed with ranges defined for corresponding codes in TfLiteSupportErrorCode. 
Logic is such that if payload can't be mapped to any valid TfLiteSupportStatus, then it will try mapping to absl status codes.